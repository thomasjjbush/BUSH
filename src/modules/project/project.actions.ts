import { GraphQLClient } from 'graphql-request';
import { Dispatch } from 'redux';
import { ProjectActions } from '../../types';
import { isImageLandscape } from '../../utils';
import { loadProjectQuery, loadSimilarQuery } from './project.queries';

export const loadProject = (graphQL: GraphQLClient, slug: string) => async (dispatch: Dispatch): Promise<void> => {
    try {
        const { projects } = await graphQL.request(loadProjectQuery, { slug });

        if (!projects.items.length) {
            dispatch({ type: ProjectActions.ERROR, payload: 404 });
            return;
        }

        const { slug: client } = projects.items[0].client;
        const { slug: tag } = projects.items[0].primaryTag;

        const { sameClient, sameTag } = await graphQL.request(loadSimilarQuery, { client, slug, tag });
        const isLandScape = await isImageLandscape(projects.items[0].gallery.items?.[0]?.url);

        dispatch({
            type: ProjectActions.LOADED_PROJECT,
            payload: {
                item: { ...projects.items[0], gallery: { ...projects.items[0].gallery, isLandScape } },
                sameClient: sameClient.items,
                sameTag: sameTag.items,
            },
        });
    } catch (err) {
        dispatch({ type: ProjectActions.ERROR, payload: 500 });
    }
};
