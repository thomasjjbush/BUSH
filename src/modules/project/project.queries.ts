import { gql } from 'graphql-request';
import { projectFragment, similarFragment } from '../../utils';

export const loadProjectQuery = gql`
    query($slug: String!) {
        projects: projectCollection(where: { slug: $slug }) {
            items {
                ...ProjectFragment
                description
                gallery: galleryCollection(limit: 10) {
                    items {
                        url
                    }
                    total
                }
                hero {
                    url
                }
                primaryTag {
                    name
                    slug
                }
                responsibilities: responsibilitiesCollection(limit: 10) {
                    items {
                        description
                        icon
                        name
                    }
                }
                video {
                    url
                }
                year
            }
        }
    }
    ${projectFragment}
`;

export const loadSimilarQuery = gql`
    query($client: String!, $slug: String!, $tag: String!) {
        sameClient: projectCollection(limit: 3, where: { client: { slug: $client }, slug_not: $slug }) {
            items {
                ...SimilarFragment
            }
        }
        sameTag: projectCollection(
            limit: 3
            where: { client: { slug_not: $client }, primaryTag: { slug: $tag }, slug_not: $slug }
        ) {
            items {
                ...SimilarFragment
                client {
                    primaryColor
                    logo {
                        url
                    }
                }
            }
        }
    }
    ${similarFragment}
`;
