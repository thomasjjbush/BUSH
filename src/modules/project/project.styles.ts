import styled, { keyframes } from 'styled-components';
import { backgroundImage } from '../../utils';
import { StyledProps } from '../../types';
import { FlexBox } from '../theme/global';
import { gradient } from '../../utils/images/images';

const animation = keyframes`
    from { opacity: 0 }
    to { opacity: 1 }
`;

export const Container = styled.div`
    margin-top: 40vh;
    background-color: ${(props): string => props.theme.colors.background};
`;

export const Description = styled.div<StyledProps>`
    grid-column: 2 / 4;

    a {
        color: ${(props): string => props.theme.colors.brand};
        text-decoration: underline;
    }

    p {
        &:first-of-type {
            margin-top: 0;
        }
        &:last-of-type {
            margin-bottom: 0;
        }
    }

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
        grid-column: auto;
        margin-top: 30px;
    }
`;

export const Grid = styled.div<StyledProps<{ cols: number; colGap?: number; rowGap?: number }>>`
    display: grid;
    grid-column-gap: ${(props): number => props.colGap || 0}px;
    grid-row-gap: ${(props): number => props.rowGap || 0}px;
    grid-template-columns: ${(props): string => `repeat(${props.cols}, minmax(0, 1fr))`};

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
        grid-template-columns: auto;
    }
`;

export const Hero = styled(FlexBox)<StyledProps<{ url: string }>>`
    background: ${({ color, theme, url }): string =>
            `${gradient(color, color + ' 80px', 'transparent')}, ${backgroundImage(url, theme.isWebpSupported, 1500)}`}
        center no-repeat;
    background-size: cover;
    flex-shrink: 0;
    height: 40vh;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;

    + div {
        padding: 0px;
    }
`;

export const Info = styled.div`
    margin: auto;
    max-width: 1200px;
    padding: 30px;
    z-index: 3;

    & > div:not(:last-child) {
        margin-bottom: 60px;
    }

    video {
        width: 100%;
    }
`;

export const Panel = styled.div<StyledProps>`
    align-self: flex-start;
    background-color: ${(props): string => props.theme.colors.offset};
    padding: 30px;
`;

export const Responsibility = styled.div<StyledProps>`
    border-bottom: solid 2px ${(props): string => props.theme.colors.brand};
    padding-bottom 15px;
    text-align: center;

    & > i {
        dispay: block;
    }

    h1 {
        text-align: center;
    }

    button {
        margin: auto;
    }
`;

export const SimilarContainer = styled(FlexBox)`
    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.tabletPortrait}px) {
        display: block;
    }
`;

export const Similar = styled.div<StyledProps<{ url: string }>>`
    background: ${({ color, theme, url }): string =>
        `${gradient('90deg', color, 'transparent')}, ${backgroundImage(url, theme.isWebpSupported)}`};
    background-size: cover;
    flex: 1;
    height: 100px;
    line-height: 100px;
    overflow: hidden;
    padding: 0 30px;

    h1 {
        color: white;
        max-width: 85%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }
`;

export const SimilarTitle = styled.p`
    margin: 30px;
`;

export const Sticky = styled(FlexBox)`
    background-color: ${(props): string => props.color};
    height: 60px;
    left: 0;
    position: fixed;
    right: 0;
    top: 60px;
    z-index: 3;

    a {
        animation: ${animation} 1s forwards;
        opacity: 0;
    }
`;

export const StickyRef = styled.div`
    height: 60px;
`;
