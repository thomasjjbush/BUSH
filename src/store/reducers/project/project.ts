import { ProjectActions, ProjectState } from '../../../types';

type Payload = number | Partial<ProjectState>;

export const projectReducer = (
    state: ProjectState = {},
    action: { payload: Payload; type: ProjectActions },
): ProjectState => {
    switch (action.type) {
        case ProjectActions.CLEARED:
            return {};
        case ProjectActions.ERROR:
            return { ...state, error: action.payload as number };
        case ProjectActions.LOADED_PROJECT:
            return { ...state, ...(action.payload as Partial<ProjectState>) };
        default:
            return state;
    }
};
