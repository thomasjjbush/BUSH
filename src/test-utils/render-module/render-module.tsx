import React, { FunctionComponent } from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { mount, ReactWrapper } from 'enzyme';
import { mockStore } from '../../test-utils/mock-store/mock-store';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../test-utils/mock-theme/mock-theme';
import { initTheme } from '../../modules/theme/theme';
import { Store } from '../../types';

const client = {
    logo: {
        url: 'client-logo-url',
    },
    name: 'client-name',
    primaryColor: 'client-primaryColor',
    slug: 'client-slug',
    url: 'client-url',
};

const project = {
    description: 'project-description',
    hero: {
        url: 'project-hero-url',
    },
    client,
    gallery: {
        isLandScape: false,
        total: 1,
        items: [
            {
                url: 'project-gallery-0-url',
            },
        ],
    },
    name: 'project-name',
    primaryTag: { name: 'primary-tag-name', slug: 'primary-tag-slug' },
    responsibilities: {
        items: [
            {
                description: 'responsibility-description',
                icon: 'responsibility-icon',
                name: 'responsibility-name',
            },
        ],
    },
    slug: 'project-slug',
    tags: { items: [{ name: 'tag-name', slug: 'tag-slug' }] },
    thumbnail: {
        name: 'x',
        url: 'project-thumbnail-url',
    },
    year: 1970,
};

const defaultState: Store = {
    clients: [client],
    employment: [
        {
            companyName: 'company',
            endDate: new Date(null).toDateString(),
            location: { lat: 'lat', lon: 'lon' },
            responsibilities: 'responsibilities',
            startDate: new Date(null).toDateString(),
            title: 'title',
            url: 'url',
        },
    ],
    project: {
        item: project,
        sameClient: [project],
        sameTag: [project],
    },
    projects: {
        items: [project],
        total: 10,
    },
    tags: [{ name: 'tag-name', slug: 'tag-slug' }],
};

export const renderModule = (Component: FunctionComponent, overrides = {}): ReactWrapper =>
    mount(
        <Provider store={mockStore({ ...defaultState, ...overrides })}>
            <ThemeProvider theme={mockTheme(initTheme(false, true) as any)}>
                <Router history={createMemoryHistory()}>
                    <Component />
                </Router>
            </ThemeProvider>
        </Provider>,
    );
