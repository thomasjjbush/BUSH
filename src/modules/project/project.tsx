import React, { Fragment, FunctionComponent, ReactElement, useCallback, useEffect, useRef, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { Carousel, Client, Error, HiddenText, Icon, Image, Scroller, Tags } from '../../components';
import { StyledMain } from '../theme/global';
import { labels } from '../../config';
import { ProjectActions, ProjectState, Store, Project as ProjectType } from '../../types';
import { loadProject } from './project.actions';
import * as Styled from './project.styles';
import { Loading } from '../../components/loading/loading';
import { useGraphQL } from '../../utils';

const renderSimilar = (title: string, items: Partial<ProjectType>[], color?: string): ReactElement => (
    <Fragment>
        <p style={{ margin: 30 }}>{title}</p>
        <Styled.SimilarContainer>
            {items.map(({ client, hero, name, slug }) => (
                <Styled.Similar color={color || client.primaryColor} key={slug} url={hero.url}>
                    <Link to={`/projects/${slug}`} data-test-id={`similar-${slug}`}>
                        <h1>{name}</h1>
                    </Link>
                </Styled.Similar>
            ))}
        </Styled.SimilarContainer>
    </Fragment>
);

const Project: FunctionComponent = (): ReactElement => {
    const graphQL = useGraphQL();
    const dispatch = useDispatch();
    const { slug } = useParams<{ slug: string }>();
    const project = useSelector<Store, ProjectState>(state => state.project);
    const [scrolled, setScrolled] = useState(0);
    const clientRef = useRef(null);
    const scrollerRef = useRef(null);
    const throttle = useRef(null);

    useEffect(() => {
        dispatch(loadProject(graphQL, slug));
        return (): void => {
            if (scrollerRef?.current?.scrollTop) {
                scrollerRef.current.scrollTop = 0;
            }
            clearTimeout(throttle.current);
            setScrolled(0);
            dispatch({ type: ProjectActions.CLEARED });
        };
    }, [slug]);

    const onScroll = useCallback(
        e => {
            if (throttle?.current) return;
            e.persist();
            throttle.current = setTimeout(() => {
                const newScroll = Math.ceil(e.target.scrollTop / 10);
                throttle.current = null;

                if (newScroll < 0) return;

                if (newScroll < 40) {
                    return setScrolled(newScroll);
                }
                if (newScroll > 40 && scrolled < 40) {
                    setScrolled(40);
                }
            }, 100);
        },
        [scrolled],
    );

    if (project.error) return <Error code={project.error} />;
    if (!project.item) return <Loading />;

    const { item, sameClient, sameTag } = project;
    const { client, description, gallery, hero, name, primaryTag, responsibilities, tags } = item;

    return (
        <StyledMain vertical>
            <Styled.Hero
                color={client.primaryColor}
                data-test-id="hero"
                style={{ height: 40 - scrolled + 'vh', minHeight: clientRef?.current?.clientHeight }}
                url={hero.url}
            >
                <Client
                    background={false}
                    logo={client.logo.url}
                    url={client.url}
                    ref={clientRef}
                    size={200}
                    width={100}
                />
            </Styled.Hero>
            <Scroller onScroll={onScroll} ref={scrollerRef}>
                <Styled.Info>
                    <Styled.Grid cols={3} colGap={30}>
                        <Styled.Panel>
                            <h1>{name}</h1>
                            <Tags tags={tags.items} />
                        </Styled.Panel>
                        <Styled.Description>{description}</Styled.Description>
                    </Styled.Grid>
                    {Boolean(gallery.total) && (
                        <Carousel
                            color={client.primaryColor}
                            margin={30}
                            numberOfItems={gallery.total}
                            itemsToDisplay={gallery.isLandScape ? 1 : 2}
                        >
                            {gallery.items.map(({ url }, i) => (
                                <Image key={i} src={url} size={800} width="100%" />
                            ))}
                        </Carousel>
                    )}
                    <Styled.Grid data-test-id="responsibilities" cols={3} colGap={30} rowGap={30}>
                        {responsibilities.items.map(({ description, icon, name }) => (
                            <Styled.Responsibility data-test-id="responsibility" key={name}>
                                <Icon color={client.primaryColor} icon={icon} size={50} />
                                <h1>{name}</h1>
                                <HiddenText lines={4}>{description}</HiddenText>
                            </Styled.Responsibility>
                        ))}
                    </Styled.Grid>
                    {project.item.video && (
                        <video style={{ width: '100%' }} controls>
                            <source src={project.item.video.url} />{' '}
                        </video>
                    )}
                </Styled.Info>
                {Boolean(sameClient?.length) &&
                    renderSimilar(labels.sameClient + client.name, sameClient, client.primaryColor)}
                {Boolean(sameTag?.length) && renderSimilar(labels.sameTag + primaryTag.name, sameTag)}
            </Scroller>
        </StyledMain>
    );
};

Project.displayName = 'Project';

export default Project;
