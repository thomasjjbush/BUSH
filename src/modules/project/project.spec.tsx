import { ReactNode } from 'react';
import { renderModule } from '../../test-utils/render-module/render-module';
import Project from './project';
import { loadProject } from './project.actions';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(() => ({ slug: 'sluggy' })),
}));

jest.mock('../../utils', () => ({
    ...jest.requireActual('../../utils'),
    useGraphQL: jest.fn().mockReturnValue('graphQLContext'),
}));

jest.mock('./project.actions', () => ({
    loadProject: jest.fn().mockReturnValue({ type: 'MOCK_LOAD_PROJECT' }),
}));

jest.mock('react-markdown', () => ({
    __esModule: true,
    default: ({ children }: { children: string }): string => children,
}));

jest.mock('./../../components', () => ({
    Carousel: ({ children }: { children: ReactNode[] }): ReactNode[] => children,
    Client: (): string => '',
    Error: (): string => '',
    HiddenText: ({ children }: { children: string }): string => children,
    Icon: (): string => '',
    Image: (): string => '',
    Scroller: ({ children }: { children: ReactNode[] }): ReactNode[] => children,
    Tags: (): string => '',
}));

describe('Project', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot', () => {
        expect(renderModule(Project).find(Project)).toMatchSnapshot();
    });

    it('should invoke loadProject on mount', () => {
        renderModule(Project);
        expect(loadProject).toHaveBeenCalledTimes(1);
        expect(loadProject).toHaveBeenCalledWith('graphQLContext', 'sluggy');
    });
});
