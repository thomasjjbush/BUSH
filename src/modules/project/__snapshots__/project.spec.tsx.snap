// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `
<Project>
  <styled.main
    styles="
    display: flex;
    flex-direction: column;
    flex: 1;
    overflow: hidden;
    padding: px;
    position: relative;
    margin: none;
    max-width: none;
    width: 100%;

    @media only screen and (max-width: theme.breakpoints.tabletPortraitpx) {
        flex-direction: column;
    }
"
    vertical={true}
  >
    <main
      className="sc-bdfBwQ ckObKr"
    >
      <Styled(styled.div)
        align="flex-end"
        color="client-primaryColor"
        data-test-id="hero"
        styles="
    display: flex;
    function (props) { return props.align && \\"align-items: \\" + props.align + \\";\\"; };
    function (props) { return props.direction && \\"flex-direction: \\" + props.direction + \\";\\"; };
    function (props) { return props.flex && \\"flex: \\" + props.flex + \\";\\"; };
    function (props) { return props.justify && \\"justify-content: \\" + props.justify + \\";\\"; };

    background: linear-gradient(client-primaryColorclient-primaryColor 80pxtransparent), url(project-hero-url?fm=webp&w=1500)
        center no-repeat;
    background-size: cover;
    flex-shrink: 0;
    height: 40vh;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;

    + div {
        padding: 0px;
    }
"
        url="project-hero-url"
      >
        <div
          className="sc-gsTCUz sc-jSgupP chyNUC gkXrpf"
          color="client-primaryColor"
          data-test-id="hero"
        >
          <styled.div
            styles="
    height: 60px;
"
          >
            <div
              className="sc-crrsfI ehEtIz"
            />
          </styled.div>
        </div>
      </Styled(styled.div)>
      <Scroller
        onScroll={[Function]}
      >
        <styled.div
          styles="
    margin-top: 40vh;
    background-color: theme.colors.background;
"
        >
          <div
            className="sc-dlfnbm gdfifr"
          >
            <styled.div
              styles="
    margin: auto;
    max-width: 1200px;
    padding: 30px;
    z-index: 3;

    & > div:not(:last-child) {
        margin-bottom: 60px;
    }

    video {
        width: 100%;
    }
"
            >
              <div
                className="sc-gKsewC bWWaRy"
              >
                <styled.div
                  colGap={30}
                  cols={3}
                  styles="
    display: grid;
    grid-column-gap: 30px;
    grid-row-gap: px;
    grid-template-columns: repeat(3, minmax(0, 1fr));

    @media only screen and (max-width: theme.breakpoints.mobilepx) {
        grid-template-columns: auto;
    }
"
                >
                  <div
                    className="sc-eCssSg fDwTkw"
                    cols={3}
                  >
                    <styled.div
                      styles="
    align-self: flex-start;
    background-color: theme.colors.offset;
    padding: 30px;
"
                    >
                      <div
                        className="sc-iBPRYJ fvlZKw"
                      >
                        <h1>
                          project-name
                        </h1>
                        <Tags
                          tags={
                            Array [
                              Object {
                                "name": "tag-name",
                                "slug": "tag-slug",
                              },
                            ]
                          }
                        />
                      </div>
                    </styled.div>
                    <styled.div
                      styles="
    grid-column: 2 / 4;

    a {
        color: theme.colors.brand;
        text-decoration: underline;
    }

    p {
        &:first-of-type {
            margin-top: 0;
        }
        &:last-of-type {
            margin-bottom: 0;
        }
    }

    @media only screen and (max-width: theme.breakpoints.mobilepx) {
        grid-column: auto;
        margin-top: 30px;
    }
"
                    >
                      <div
                        className="sc-hKgILt hfaMQT"
                      >
                        <default>
                          project-description
                        </default>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <Carousel
                  color="client-primaryColor"
                  itemsToDisplay={2}
                  margin={30}
                  numberOfItems={1}
                >
                  <Image
                    key="0"
                    size={800}
                    src="project-gallery-0-url"
                    width="100%"
                  />
                </Carousel>
                <styled.div
                  colGap={30}
                  cols={3}
                  data-test-id="responsibilities"
                  rowGap={30}
                  styles="
    display: grid;
    grid-column-gap: 30px;
    grid-row-gap: 30px;
    grid-template-columns: repeat(3, minmax(0, 1fr));

    @media only screen and (max-width: theme.breakpoints.mobilepx) {
        grid-template-columns: auto;
    }
"
                >
                  <div
                    className="sc-eCssSg jdUAfz"
                    cols={3}
                    data-test-id="responsibilities"
                  >
                    <styled.div
                      data-test-id="responsibility"
                      key="responsibility-name"
                      styles="
    border-bottom: solid 2px theme.colors.brand;
    padding-bottom 15px;
    text-align: center;

    & > i {
        dispay: block;
    }

    h1 {
        text-align: center;
    }

    button {
        margin: auto;
    }
"
                    >
                      <div
                        className="sc-fubCfw JjOvC"
                        data-test-id="responsibility"
                      >
                        <Icon
                          color="client-primaryColor"
                          icon="responsibility-icon"
                          size={50}
                        />
                        <h1>
                          responsibility-name
                        </h1>
                        <HiddenText
                          lines={4}
                        >
                          responsibility-description
                        </HiddenText>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </div>
            </styled.div>
            <styled.p
              styles="
    margin: 30px;
"
            >
              <p
                className="sc-kEjbxe cyWVKX"
              >
                laebls.sameClientclient-name
              </p>
            </styled.p>
            <Styled(styled.div)
              styles="
    display: flex;
    function (props) { return props.align && \\"align-items: \\" + props.align + \\";\\"; };
    function (props) { return props.direction && \\"flex-direction: \\" + props.direction + \\";\\"; };
    function (props) { return props.flex && \\"flex: \\" + props.flex + \\";\\"; };
    function (props) { return props.justify && \\"justify-content: \\" + props.justify + \\";\\"; };

    @media only screen and (max-width: theme.breakpoints.tabletPortraitpx) {
        display: block;
    }
"
            >
              <div
                className="sc-gsTCUz sc-pFZIQ clQfFA bYkmA"
              >
                <styled.div
                  color="client-primaryColor"
                  key="project-slug"
                  styles="
    background: linear-gradient(90degclient-primaryColortransparent), url(project-hero-url?fm=webp&w=750);
    background-size: cover;
    flex: 1;
    height: 100px;
    line-height: 100px;
    overflow: hidden;
    padding: 0 30px;

    h1 {
        color: white;
        max-width: 85%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }
"
                  url="project-hero-url"
                >
                  <div
                    className="sc-jrAGrp dstIzR"
                    color="client-primaryColor"
                  >
                    <Link
                      data-test-id="similar-project-slug"
                      to="/projects/project-slug"
                    >
                      <LinkAnchor
                        data-test-id="similar-project-slug"
                        href="/projects/project-slug"
                        navigate={[Function]}
                      >
                        <a
                          data-test-id="similar-project-slug"
                          href="/projects/project-slug"
                          onClick={[Function]}
                        >
                          <h1>
                            project-name
                          </h1>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                </styled.div>
              </div>
            </Styled(styled.div)>
            <styled.p
              styles="
    margin: 30px;
"
            >
              <p
                className="sc-kEjbxe cyWVKX"
              >
                laebls.sameTagprimary-tag-name
              </p>
            </styled.p>
            <Styled(styled.div)
              styles="
    display: flex;
    function (props) { return props.align && \\"align-items: \\" + props.align + \\";\\"; };
    function (props) { return props.direction && \\"flex-direction: \\" + props.direction + \\";\\"; };
    function (props) { return props.flex && \\"flex: \\" + props.flex + \\";\\"; };
    function (props) { return props.justify && \\"justify-content: \\" + props.justify + \\";\\"; };

    @media only screen and (max-width: theme.breakpoints.tabletPortraitpx) {
        display: block;
    }
"
            >
              <div
                className="sc-gsTCUz sc-pFZIQ clQfFA bYkmA"
              >
                <styled.div
                  color="client-primaryColor"
                  key="project-slug"
                  styles="
    background: linear-gradient(90degclient-primaryColortransparent), url(project-hero-url?fm=webp&w=750);
    background-size: cover;
    flex: 1;
    height: 100px;
    line-height: 100px;
    overflow: hidden;
    padding: 0 30px;

    h1 {
        color: white;
        max-width: 85%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }
"
                  url="project-hero-url"
                >
                  <div
                    className="sc-jrAGrp dstIzR"
                    color="client-primaryColor"
                  >
                    <Link
                      data-test-id="similar-project-slug"
                      to="/projects/project-slug"
                    >
                      <LinkAnchor
                        data-test-id="similar-project-slug"
                        href="/projects/project-slug"
                        navigate={[Function]}
                      >
                        <a
                          data-test-id="similar-project-slug"
                          href="/projects/project-slug"
                          onClick={[Function]}
                        >
                          <h1>
                            project-name
                          </h1>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                </styled.div>
              </div>
            </Styled(styled.div)>
          </div>
        </styled.div>
      </Scroller>
    </main>
  </styled.main>
</Project>
`;
